k8s/services/grafana/helm-8.5.cue

```diff
@@ -2,436 +2,122 @@ package grafana
-		podPortName: string | *"http"
-		revisionHistoryLimit: int | *10
-		networkPolicy: {
-			limitIngress: bool | *true
-			limitEgress: bool | *false
-			extraIngressSelectors: [...{}] | *[]
-		}
-		resources: {} | *{}
-		nodeSelector: {} | *{}
-		tolerations: [...{}] | *[]
-		affinity: {} | *{}
-		extraConfigmapMounts: [...{}] | *[]
-		extraSecretMounts: [...{}] | *[]
-		extraVolumeMounts: [...{}] | *[]
-		extraVolumes: [...{}] | *[]
-	}
-	networkPolicy: {
-		enabled: bool | *false
-		ingress: bool | *true
-		allowExternal: bool | *true
-		explicitNamespacesSelector: {} | *{}
-		egress: {
-			enabled: bool | *false
-			blockDNSResolution: bool | *false
-			ports: [...{}] | *[]
-			to: [...{}] | *[]
-		}
-	}
-	enableKubeBackwardCompatibility: bool | *false
-	useStatefulSet: bool | *false
-	extraObjects: [...{}] | *[]
-	assertNoLeakedSecrets: bool | *true
+    global: {
+        imageRegistry: string | *null
+        imagePullSecrets: [...string] | *[]
+    }
+
+    grafana: {
+        enabled:      bool | *true
+        isDefault:    bool | *true
+
+        // Health check probes
+        readinessProbe: {
+            httpGet: {
+                path: string | *"/api/health"
+                port: int | *3000
+            }
+        }
+        livenessProbe: {
+            httpGet: {
+                path: string | *"/api/health"
+                port: int | *3000
+            }
+            initialDelaySeconds: int | *60
+            timeoutSeconds:      int | *30
+            failureThreshold:    int | *10
+        }
+
+        // Service configuration
+        service: {
+            enabled:      bool | *true
+            type:         string | *"ClusterIP"
+            port:         int | *80
+            targetPort:   int | *3000
+            annotations:  k8s.#Annotations
+            labels:       k8s.#Labels
+        }
+
+        // Image configuration for Grafana
+        image: {
+            registry:     string | *"docker.io"
+            repository:   string | *"grafana/grafana"
+            tag:          string | *"latest"
+            pullPolicy:   v1.PullPolicy | *"IfNotPresent"
+        }
+
+        // Resource requests and limits
+        resources: {
+            limits: {
+                cpu:    string | *"500m"
+                memory: string | *"1Gi"
+            }
+            requests: {
+                cpu:    string | *"250m"
+                memory: string | *"512Mi"
+            }
+        }
+
+        // Persistence settings
+        persistence: {
+            enabled:     bool | *false
+            accessModes: [...string] | *["ReadWriteOnce"]
+            size:        string | *"10Gi"
+        }
+
+        // Additional configurations for pods
+        nodeSelector: k8s.#Labels
+        tolerations:  [...v1.#Toleration]
+        affinity:     v1.#Affinity
+    }
+
+    // Datasources for Grafana
+    datasources: {
+        "datasources.yaml": {
+            apiVersion: 1
+            datasources: [
```

Comment:
this field cannot be overriden, CUE will prevent this, you should define a schema instead and add these data sources as the default values

Reply:
to test this try installing the observability stack in a namespace other than "monitoring", you'll need then to override the data source URLs to use the different namespace but will not be able to do this